scripts/ci/test_plan.py

```diff
@@ -303,8 +364,6 @@ def find_tests(self):
             if self.platforms:
                 for platform in self.platforms:
                     _options.extend(["-p", platform])
-            else:
-                _options.append("--all")
```

Comment:
this is quite a major change in the functioning beyond cleanup/refactoring. I am fine with it but just pointing out that that it is hard to catch and nowhere documented.
Before, when a test was changed, a CI was supposed to run on all available platforms. Now those will be only the default ones. IMO integration platforms (and if not added then defautl ones) could make more sense